src/explorer/AzureAccountTreeItem.ts

```diff
@@ -2,17 +2,203 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
+import { UiStrings } from "../uiStrings";
+import * as vscode from "vscode";
+import { AzExtParentTreeItem, AzExtTreeItem, registerEvent, GenericTreeItem, ISubscriptionContext } from '@microsoft/vscode-azext-utils';
+import { AzureSubscriptionHelper } from "../azure/azureLogin/subscriptions";
+import { AzureSessionProvider, ReadyAzureSessionProvider, SelectionType, SignInStatus } from "../azure/azureLogin/authTypes";
+import { GeneralUtils } from "../utils/generalUtils";
+import { AzureAuth } from "../azure/azureLogin/azureAuth";
+import { APIMAccountCommandId } from "../constants";
+import { Subscription } from "@azure/arm-resources-subscriptions";
+import { createSubscriptionTreeItem } from "./ApiManagementProvider";
 
-import { ISubscriptionContext } from '@microsoft/vscode-azext-utils';
-import { AzureAccountTreeItemBase } from '@microsoft/vscode-azext-azureutils';
-import { ApiManagementProvider } from './ApiManagementProvider';
+export function createAzureAccountTreeItem(
+    sessionProvider: AzureSessionProvider,
+): AzExtParentTreeItem & { dispose(): unknown } {
+    return new AzureAccountTreeItem(sessionProvider);
```

Comment:
Can we call the constructor directly in extension.ts?

Reply:
Here I just want to keep the tree parent clear, since the `createAzureAccountTreeItem` return type is `AzExtParentTreeItem & { dispose(): unknown }`, and the barely `AzureAccountTreeItem` extends `AzExtParentTreeItem`. so just create the create method.

Reply:
The vscode-azext-azureutils is declaring the class as `class AzureAccountTreeItemBase extends AzExtParentTreeItem implements Disposable`. Can we do similar things? It seems fine to implement Disposable if VS Code is expecting a disposable tree item. And we don't need to convert the type through this function by doing so.

Reply:
updated